internal/gps/pkgtree/digest_test.go

```diff
@@ -0,0 +1,187 @@
+			t.Errorf("Input: %#v; (GOT): %#q; (WNT): %#q", testCase.input, got, want)
+		}
+	}
+}
+
+////////////////////////////////////////
+
+func getTestdataVerifyRoot(t *testing.T) string {
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	return filepath.Join(filepath.Dir(cwd), "testdata_digest")
+}
+
+func TestDigestFromDirectoryBailsUnlessDirectory(t *testing.T) {
+	prefix := getTestdataVerifyRoot(t)
+	relativePathname := "launchpad.net/match"
+	_, err := DigestFromDirectory(filepath.Join(prefix, relativePathname))
+	if got, want := err, error(nil); got != want {
+		t.Errorf("\n(GOT): %v; (WNT): %v", got, want)
+	}
+}
+
+func TestDigestFromDirectory(t *testing.T) {
+	relativePathname := "launchpad.net/match"
+	want := []byte{0x7e, 0x10, 0x6, 0x2f, 0x8, 0x3, 0x3c, 0x76, 0xae, 0xbc, 0xa4, 0xc9, 0xec, 0x73, 0x67, 0x15, 0x70, 0x2b, 0x0, 0x89, 0x27, 0xbb, 0x61, 0x9d, 0xc7, 0xc3, 0x39, 0x46, 0x3, 0x91, 0xb7, 0x3b}
+
+	// NOTE: Create the hash using both an absolute and a relative pathname to
+	// ensure hash ignores prefix.
+
+	t.Run("AbsolutePrefix", func(t *testing.T) {
+		prefix := getTestdataVerifyRoot(t)
+		got, err := DigestFromDirectory(filepath.Join(prefix, relativePathname))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !bytes.Equal(got, want) {
+			t.Errorf("\n(GOT):\n\t%#v\n(WNT):\n\t%#v", got, want)
+		}
+	})
+
+	t.Run("RelativePrefix", func(t *testing.T) {
+		prefix := "../testdata_digest"
+		got, err := DigestFromDirectory(filepath.Join(prefix, relativePathname))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !bytes.Equal(got, want) {
+			t.Errorf("\n(GOT):\n\t%#v\n(WNT):\n\t%#v", got, want)
+		}
+	})
+}
+
+func TestVerifyDepTree(t *testing.T) {
+	vendorRoot := getTestdataVerifyRoot(t)
+
+	wantSums := map[string][]byte{
+		"github.com/alice/match":       []byte{0x7e, 0x10, 0x6, 0x2f, 0x8, 0x3, 0x3c, 0x76, 0xae, 0xbc, 0xa4, 0xc9, 0xec, 0x73, 0x67, 0x15, 0x70, 0x2b, 0x0, 0x89, 0x27, 0xbb, 0x61, 0x9d, 0xc7, 0xc3, 0x39, 0x46, 0x3, 0x91, 0xb7, 0x3b},
+		"github.com/alice/mismatch":    []byte("some non-matching digest"),
+		"github.com/bob/emptyDigest":   nil, // empty hash result
+		"github.com/bob/match":         []byte{0x7e, 0x10, 0x6, 0x2f, 0x8, 0x3, 0x3c, 0x76, 0xae, 0xbc, 0xa4, 0xc9, 0xec, 0x73, 0x67, 0x15, 0x70, 0x2b, 0x0, 0x89, 0x27, 0xbb, 0x61, 0x9d, 0xc7, 0xc3, 0x39, 0x46, 0x3, 0x91, 0xb7, 0x3b},
+		"github.com/charlie/notInTree": nil, // not in tree result ought to superseede empty digest result
+		// matching result at seldomly found directory level
+		"launchpad.net/match": []byte{0x7e, 0x10, 0x6, 0x2f, 0x8, 0x3, 0x3c, 0x76, 0xae, 0xbc, 0xa4, 0xc9, 0xec, 0x73, 0x67, 0x15, 0x70, 0x2b, 0x0, 0x89, 0x27, 0xbb, 0x61, 0x9d, 0xc7, 0xc3, 0x39, 0x46, 0x3, 0x91, 0xb7, 0x3b},
+	}
+
+	status, err := VerifyDepTree(vendorRoot, wantSums)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// NOTE: When true, display the digests of the directories specified by the
+	// digest keys.
+	if false {
```

Comment:
i assume this is here for debugging purposes? why not put it down below and have the conditional be on `t.Failed()`, so that the mismatch will automatically be printed in the event of a failure?